pinot-spi/src/main/java/org/apache/pinot/spi/trace/RequestContext.java

```diff
@@ -157,6 +161,58 @@ default boolean isSampledRequest() {
 
   void setReduceTimeMillis(long reduceTimeMillis);
 
+  long getNumConsumingSegmentsQueried();
+
+  void setNumConsumingSegmentsQueried(long numConsumingSegmentsQueried);
+
+  long getNumConsumingSegmentsProcessed();
+
+  void setNumConsumingSegmentsProcessed(long numConsumingSegmentsProcessed);
+
+  long getNumConsumingSegmentsMatched();
+
+  void setNumConsumingSegmentsMatched(long numConsumingSegmentsMatched);
+
+  long getMinConsumingFreshnessTimeMs();
+
+  void setMinConsumingFreshnessTimeMs(long minConsumingFreshnessTimeMs);
+
+  long getNumSegmentsPrunedByBroker();
+
+  void setNumSegmentsPrunedByBroker(long numSegmentsPrunedByBroker);
+
+  long getNumSegmentsPrunedByServer();
+
+  void setNumSegmentsPrunedByServer(long numSegmentsPrunedByServer);
+
+  long getNumSegmentsPrunedInvalid();
+
+  void setNumSegmentsPrunedInvalid(long numSegmentsPrunedInvalid);
+
+  long getNumSegmentsPrunedByLimit();
+
+  void setNumSegmentsPrunedByLimit(long numSegmentsPrunedByLimit);
+
+  long getNumSegmentsPrunedByValue();
+
+  void setNumSegmentsPrunedByValue(long numSegmentsPrunedByValue);
+
+  long getExplainPlanNumEmptyFilterSegments();
+
+  void setExplainPlanNumEmptyFilterSegments(long explainPlanNumEmptyFilterSegments);
+
+  long getExplainPlanNumMatchAllFilterSegments();
+
+  void setExplainPlanNumMatchAllFilterSegments(long explainPlanNumMatchAllFilterSegments);
+
+  Map<String, String> getTraceInfo();
+
+  void setTraceInfo(Map<String, String> traceInfo);
```

Comment:
i am not sure default creating these for tracing is a good idea. normally we dont enable tracing for production use case, so for those that were manually enabled, those must've been monitored and thus renders it less useful to keep the event listener (e.g. there's a human tracking it already)

Reply:
Yes but we would still want to keep the trace-info in the event so that we can persist it in Kafka / Hive and refer it later.